src/v/redpanda/application.cc

```diff
@@ -600,7 +600,15 @@ ss::app_template::config application::setup_app_config() {
 }
 
 void application::hydrate_config(const po::variables_map& cfg) {
-    std::filesystem::path cfg_path(cfg["redpanda-cfg"].as<std::string>());
+    auto raw_cfg_path = cfg["redpanda-cfg"].as<std::string>();
+    // Expand ~/redpanda.yaml to the full path
+    if (raw_cfg_path.starts_with("~")) {
+        const char* home = std::getenv("HOME");
+        if (home) {
+            raw_cfg_path = fmt::format("{}{}", home, raw_cfg_path.substr(1));
+        }
+    }
```

Comment:
did you consider using `wordexp` which I think handles tilde as well as other expansions that generally work in a shell.

Reply:
I did not know about that libc function, but it looks like it will even run arbitrary commands, which doesn't sound very fun.

This is similar to what we do for the data directory. 

https://github.com/redpanda-data/redpanda/blob/3b50a0674fb27a34caa30ac35b79957850ab18d3/src/v/config/data_directory_path.h#L53

Reply:
hah ok sounds sgtm to not use wordexp then